solr/core/src/java/org/apache/solr/util/PackageTool.java

```diff
@@ -0,0 +1,255 @@
+                break;
+              }
+              case "help":
+              case "usage":
+                System.out.println("./solr package add-repo <repository-name> <repository-url>");
+                System.out.println("Add a repository to Solr.");
+                System.out.println("");
+                System.out.println("./solr package install <package-name>[:<version>] ");
+                System.out.println("Install a package into Solr. This copies over the artifacts from the repository into Solr's internal package store and sets up classloader for this package to be used.");
+                System.out.println("");
+                System.out.println("./solr package deploy <package-name>[:<version>] [-y] [--update] -collections <comma-separated-collections> [-p <param1>=<val1> -p <param2>=<val2> ...] ");
+                System.out.println("Bootstraps a previously installed package into the specified collections. It the package accepts parameters for its setup commands, they can be specified (as per package documentation).");
+                System.out.println("");
+                System.out.println("./solr package list-installed");
+                System.out.println("Print a list of packages installed in Solr.");
+                System.out.println("");
+                System.out.println("./solr package list-available");
+                System.out.println("Print a list of packages available in the repositories.");
+                System.out.println("");
+                System.out.println("./solr package list-deployed -c <collection>");
+                System.out.println("Print a list of packages deployed on a given collection.");
+                System.out.println("");
+                System.out.println("./solr package list-deployed <package-name>");
+                System.out.println("Print a list of collections on which a given package has been deployed.");
+                System.out.println("");
+                System.out.println("./solr package undeploy <package-name> -collections <comma-separated-collections>");
+                System.out.println("Undeploys a package from specified collection(s)");
+                break;
+              default:
+                throw new RuntimeException("Unrecognized command: "+cmd);
+            };
+          } finally {
+            packageManager.close();
+          }
+        }
+      }
+      log.info("Finished: "+cmd);
+
+    } catch (Exception ex) {
+      ex.printStackTrace(); // We need to print this since SolrCLI drops the stack trace in favour of brevity. Package tool should surely print full stacktraces!
+      throw ex;
+    }
+  }
+
+  /**
+   * Parses package name and version in the format "name:version" or "name"
+   * @param arg User supplied argument
+   * @return Array of two elements, first the package name, second version (or null if not present)
+   */
+  private String[] parsePackageVersion(String arg) {
+    String packageName = arg.split(":")[0];
```

Comment:
Check for extra `:` as well, i.e. `foo:1.2.3:bar` is not allowed? Could just as well validate the version part as a legal version too.

I'm not very fond of return type `String[]` where the caller must know some magic. Consider `Pair<String,com.github.zafarkhaja.semver.Version>` as a more clear return type.

Reply:
I've fixed it in a subsequent commit.